scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/RuntimeReducerEstimator.scala

```diff
@@ -0,0 +1,107 @@
+package com.twitter.scalding.reducer_estimation
+
+import scala.collection.JavaConverters._
+import cascading.flow.FlowStep
+import cascading.tap.{ Tap, CompositeTap }
+import cascading.tap.hadoop.Hfs
+import org.apache.hadoop.mapred.JobConf
+import org.slf4j.LoggerFactory
+
+object RuntimeReducerEstimator {
+  val RuntimePerReducer = "scalding.reducer.estimator.runtime.per.reducer"
```

Comment:
runtime per reducer sounds strange. This means if you have `n` reducers you expect `n * time` to finish?

I think a good approach is to set a goal time for how long the reducers run, say 10 minutes. Then we look at the history and get a sequence of pairs: (reducers, time-required). We could take the 80%-ile time to finish instead of the actual maximum (to deal with the case of skew). Finally, we can make least squares fit that gives us the number of reducers we should choose to reach the goal time.